cmake/PureCMakeBuild.cmake

```diff
@@ -0,0 +1,290 @@
+endif ()
+
+set(SHARED_LIBS FALSE CACHE BOOL "Set to TRUE to build shared library")
+if (SHARED_LIBS)
+	add_library(${LIB_NAME} SHARED ${NABO_SRC})
+	install(TARGETS ${LIB_NAME} LIBRARY DESTINATION lib)
+else ()
+	add_library(${LIB_NAME} STATIC ${NABO_SRC})
+	if (NOT MSVC)
+		target_compile_options(${LIB_NAME} PRIVATE -fPIC)
+	endif()
+	install(TARGETS ${LIB_NAME} ARCHIVE DESTINATION lib)
+endif ()
+set_target_properties(${LIB_NAME} PROPERTIES VERSION "${PROJECT_VERSION}" SOVERSION 1)
+
+target_include_directories(${LIB_NAME} PUBLIC
+	${EIGEN_INCLUDE_DIR}
+	$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+  )
+
+# openmp
+set(USE_OPEN_MP TRUE CACHE BOOL "Set to FALSE to not use OpenMP")
+if (USE_OPEN_MP)
+	find_package(OpenMP)
+	if (OPENMP_FOUND)
+		target_compile_options(${LIB_NAME} PRIVATE -fopenmp ${OpenMP_CXX_FLAGS})
+		target_compile_definitions(${LIB_NAME} PRIVATE HAVE_OPENMP)
+		if (CMAKE_COMPILER_IS_GNUCC)
+			target_link_libraries(${LIB_NAME} PUBLIC gomp)
+		endif()
+	endif()
+endif ()
+
+
+# create doc before installing
+set(DOC_INSTALL_TARGET "share/doc/${PROJECT_NAME}/api" CACHE STRING "Target where to install doxygen documentation")
+install(FILES nabo/nabo.h DESTINATION include/nabo)
+install(FILES nabo/third_party/any.hpp DESTINATION include/nabo/third_party)
+install(FILES README.md DESTINATION share/doc/${PROJECT_NAME})
+if (DOXYGEN_FOUND)
+	install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/doc/html DESTINATION ${DOC_INSTALL_TARGET})
+endif (DOXYGEN_FOUND)
+
+enable_testing()
+
+
+option(LIBNABO_BUILD_EXAMPLES "Build libnabo examples" ON)
+if (LIBNABO_BUILD_EXAMPLES)
+  add_subdirectory(examples)
+endif()
+
+option(LIBNABO_BUILD_TESTS "Build libnabo tests" ON)
+if(LIBNABO_BUILD_TESTS)
+  add_subdirectory(tests)
+endif()
+
+option(LIBNABO_BUILD_PYTHON "Build libnabo python" ON)
+if(LIBNABO_BUILD_PYTHON)
+  add_subdirectory(python)
+endif()
+
+# Install catkin package.xml
+install(FILES package.xml DESTINATION share/libnabo)
+
+#=============================================
+# to allow find_package() on libnabo
+#=============================================
+#
+# the following case be used in an external project requiring libnabo:
+#  ...
+#  find_package(libnabo)
+#  include_directories(${libnabo_INCLUDE_DIRS})
+#  target_link_libraries(executableName ${libnabo_LIBRARIES})
+#  ...
+
+# NOTE: the following will support find_package for 1) local build (make) and 2) for installed files (make install)
+
+# 1- local build #
+
+# Register the local build in case one doesn't use "make install"
+export(PACKAGE libnabo)
+
+# 'make install' to the correct locations (provided by GNUInstallDirs).
+install(TARGETS ${LIB_NAME} EXPORT ${PROJECT_NAME}-targets
+    ARCHIVE  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME  DESTINATION ${CMAKE_INSTALL_BINDIR})  # This is for Windows
+#install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
```

Comment:
Why is it here? If it was replaced with the above single-file install commands, then delete this.